### PR TITLE
fix: fix flaky report test

### PR DESCRIPTION
**Pull request by emilyrohrbough**
- Apart of #18822 

https://dashboard.cypress.io/projects/ypt4pf/runs/31638/overview/0296e264-b0b3-4ebc-997e-a0e473f8fdfa
Screen Shot 2022-04-01 at 11 34 27 AM
Screen Shot 2022-04-01 at 11 35 05 AM

### CODE REVIEW

**Review by tbiethman — DISMISSED** [review]

**Review by davidmunechika — DISMISSED** [review]
lgtm

File: packages/driver/src/cypress.ts [hunk 1]
```diff
@@ -555,7 +555,9 @@ class $Cypress {
         return this.emit('log:added', ...args)
 
       case 'command:log:changed':
-        this.runner.addLog(args[0], this.config('isInteractive'))
+        // Cypress logs will only trigger an update every 4 seconds so there is a
```
Comment (tgriesser) [inline packages/driver/src/cypress.ts:558]:
```suggestion
        // Cypress logs will only trigger an update every 4 milliseconds so there is a
```

File: packages/runner-shared/src/event-manager.js [hunk 2]
```diff
@@ -431,11 +431,15 @@ export const eventManager = {
     })
 
     Cypress.on('log:changed', (log) => {
-      const displayProps = Cypress.runner.getDisplayPropsForLog(log)
+      // Cypress logs will only trigger an update every 4 seconds so there is a
```
Comment (tgriesser) [inline packages/runner-shared/src/event-manager.js:434]:
```suggestion
      // Cypress logs will only trigger an update every 4 milliseconds so there is a
```

**Comment by tgriesser** [comment]
Was looking into this to understand a bit better, and I think this flake actually might actually be due to a legitimate regression in the `chore: refactor log function to class` merge here: https://github.com/cypress-io/cypress/pull/20382 

Specifically the fact that the original behavior seemed to create a single debounced `fireChangeEvent` per Log class.

https://github.com/cypress-io/cypress/blob/047aefc4ba0b2eeb60b869e8ac2fab1322034fa8/packages/driver/src/cypress/log.ts#L552-L556

whereas the new behavior:

https://github.com/sainthkh/cypress/blob/ce95894a35bedbee51920ae7073ee48a241195a4/packages/driver/src/cypress/log.ts#L556-L566

appears to create a new debounced fn, which is immediately invoked, per event fired. Might be missing something non-obvious since there's quite a few `fn` binds, etc. going on here, but that's just how it appears at first glance.

/cc @sainthkh

**Review by tgriesser — CHANGES_REQUESTED** [review]

**Comment by brian-mann** [comment]
Yes this is the problem per @tgriesser comment:

https://github.com/cypress-io/cypress/pull/20886#issuecomment-1088120615

The debounce cannot be on each log instance - it needs to be a singleton and all log calls need to go through that function.

**Comment by sainthkh** [comment]
I admit that I refactored it wrong way. I didn't read the `_.debounce()` carefully. [It says that](https://lodash.com/docs/4.17.15#debounce):

```
Subsequent calls to the debounced function return the result of the last func invocation.
```

I guess creating debounce function every call changed the behavior. 

----

As `fireChangeEvent` is an internal function only used inside `log.ts` file. I think it's OK to give it `any` type. 

As I don't have the authority to push commits to this PR. I left my code suggestion at #20908.

**Review by tgriesser — APPROVED** [review]

**Review by davidmunechika — APPROVED** [review]

**Pull request merged**